planner/planner.go

```diff
@@ -280,67 +291,130 @@ func (p *MyPlanner) cityStatsHandler(context *gin.Context) {
 	context.JSON(http.StatusOK, view)
 }
 
-func (p *MyPlanner) Planning(ctx context.Context, planningRequest *PlanningReq, user string) (resp PlanningResponse) {
-	planningResponse := p.Solver.Solve(ctx, planningRequest)
+func (p *MyPlanner) Planning(ctx context.Context, planningRequest *PlanningRequest, user string) (resp PlanningResponse) {
+	logger := iowrappers.Logger
+	var planningResponse *PlanningResp
+	planningResponse = p.Solver.Solve(ctx, planningRequest)
+	resp = p.processPlanningResp(ctx, planningRequest, planningResponse, user)
+	if !planningRequest.WithNearbyCities {
+		return resp
+	}
 
-	if planningResponse.Err != nil {
-		resp.Err = planningResponse.Err
-		resp.StatusCode = planningResponse.ErrorCode
-		return
+	// prioritize planning results from the primary location
+	if resp.Err == nil && len(resp.TravelPlans) >= planningRequest.NumPlans {
+		return resp
+	}
+
+	var lat, lng float64
```

Comment:
nit: combine declaration and assignment into one line?

Reply:
Done